### DISCUSSION TITLE
ETA for 3.0?

### DISCUSSION THREAD

**Discussion by nikhiltri** [area17/twill#1432] [General]
Hello,

Do you have an ETA for when you expect 3.0 to be released? With Laravel 9.0 being released today, we're trying to plan for when we should do our next round of upgrades.

Thanks!
nikhil

**Comment by kallefrombosnia** [comment]
L9 will be supported in the 2.8.0 version of Twill (https://github.com/area17/twill/projects/3#card-77245829).
As of current 2.6.0, 2.7.0 should come soon since most of the PR and issues in that roadmap are resolved.
But this is totally my opinion, more of info @ifox could provide.

**Comment by ifox** [comment]
Hi @nikhiltri! 

Thanks for asking, and thanks @kallefrombosnia for identifying that in the [new public roadmap project](https://github.com/area17/twill/projects/3)!

We're still working on it before communicating more actively about it, but there are already clear steps ahead that should help your decision:

- Laravel 9 support is going to be released on Twill 2 this month
- Twill 3 will be released later this year and will only support PHP 8+ and Laravel 9+ (potentially Laravel 8 too, but on PHP 8+, not 7.4)
- Twill 3 will introduce breaking changes for which we are going to offer a https://github.com/rectorphp/rector based upgrade script. You may want to look into Rector to upgrade your entire application for PHP 8
- We will continue supporting Twill 2 regarding bug and security issues in 2022. Towards the end of the year, PHP and Laravel versions won't be supported against security issues anymore, so that will be the case for Twill 2 as well.

I hope that's helpful.

**Reply by nikhiltri** [reply]
That's helpful. Thank you!

**Comment by haringsrob** [comment]
While there is no release yet you can see the biggest changes and instructions here: https://github.com/area17/twill/pull/1481

**Reply by nikhiltri** [reply]
Thank you!!
